### PR TITLE
build: pin numpy build version correctly

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ pyink-indentation = 2
 pyink-use-majority-quotes = true
 
 [build-system]
-requires = ["setuptools>=61", "numpy~=1.21", "pybind11>=2.10.3"]
+requires = ["setuptools>=61", "numpy~=1.21.0", "pybind11>=2.10.3"]
```
Comment (hawkinsp) [inline pyproject.toml:44]: Should we pin the other deps also?

**Review by hawkinsp — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (jakevdp) [reply pyproject.toml]: I think these are not as important as getting numpy right, because there's no API compatibility concern.

Comment (hawkinsp) [reply pyproject.toml]: Sure. I just figured there was no harm in being more aggressive with our pins to prevent breakage on new releases.

**Review by hawkinsp — APPROVED** [review]

**Comment by jakevdp** [comment]
Replaced by #34

**Pull request closed without merging**